src/nspawn/nspawn-cgroup.c

```diff
@@ -100,14 +111,23 @@ int sync_cgroup(pid_t pid, CGroupUnified unified_requested) {
 
         undo_mount = true;
 
+        fn = strjoina(tree, cgroup);
+        (void) rm_rf(fn, REMOVE_ROOT|REMOVE_ONLY_DIRECTORIES);
```

Comment:
Hm, this strange.
Did you run `^]]]`? . i.e.
```
Container master terminated by signal KILL.
```
What is the output of the
```
mkdir cgroup
mount -t cgroup cgroup -o none,name=systemd,xattr cgroup
ls -lR cgroup/
```
after the first run of the `systemd-nspawn -D ... -b` (without `-U`)
?